packages/smooth_app/lib/pages/all_product_list_page.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/data_models/product_list.dart';
+import 'package:smooth_app/database/dao_product_list.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/pages/preferences/user_preferences_list_tile.dart';
+import 'package:smooth_app/pages/product/common/product_query_page_helper.dart';
+import 'package:smooth_app/pages/product_list_user_dialog_helper.dart';
+import 'package:smooth_app/widgets/smooth_app_bar.dart';
+import 'package:smooth_app/widgets/smooth_scaffold.dart';
+
+/// Page that lists all product lists.
+class AllProductListPage extends StatefulWidget {
+  const AllProductListPage();
+
+  @override
+  State<AllProductListPage> createState() => _AllProductListPageState();
+}
+
+class _AllProductListPageState extends State<AllProductListPage> {
+  @override
+  Widget build(BuildContext context) {
+    final LocalDatabase localDatabase = context.watch<LocalDatabase>();
```

Comment:
You are in a StatefulWidget and you force the list to be refreshed on every build.
That's unnecessary I think. Loading it an attribute of the class and reloading only when necessary would be better.

But actually I do wonder why you need a `Stateful` here, as you never call the `setState` method